addon/components/polaris-resource-list.js

```diff
@@ -1,34 +1,797 @@
+        'filterControl',
+        'itemsExist'
+      );
+
+      return filterControl && !itemsExist;
+    }).readOnly(),
+
+    spinnerStyle: computed('loadingPosition', function() {
+      let loadingPosition = this.get('loadingPosition');
+      let defaultTopPadding = 8;
+      let topPadding =
+        loadingPosition > 0 ? loadingPosition : defaultTopPadding;
+      return htmlSafe(`padding-top: ${topPadding}px`);
+    }).readOnly(),
+
+    spinnerSize: computed('items.length', function() {
+      return this.get('items.length') === 1 ? 'small' : 'large';
+    }).readOnly(),
+
+    context: computed(
+      'selectedItems.[]',
+      'resourceName',
+      'loading',
+      'selectMode',
+      'selectable',
+      function() {
+        let {
+          selectedItems,
+          resourceName,
+          loading,
+          selectMode,
+          selectable,
+        } = this.getProperties(
+          'selectedItems',
+          'resourceName',
+          'loading',
+          'selectMode',
+          'selectable'
+        );
+        resourceName = resourceName || this.get('defaultResourceName');
+        return {
+          selectable,
+          selectedItems,
+          selectMode,
+          resourceName,
+          loading,
+          onSelectionChange: (...args) => {
+            return this.handleSelectionChange(...args);
+          },
+        };
+      }
+    ).readOnly(),
+
+    handleResize() {
+      // This is needed to replicate the React implementation's `@debounce` decorator.
+      this.debounceTask('debouncedHandleResize', 0);
+    },
+
+    debouncedHandleResize() {
+      let { selectedItems, selectMode } = this.getProperties(
+        'selectedItems',
+        'selectMode'
+      );
+
+      if (
+        selectedItems &&
+        selectedItems.length === 0 &&
+        selectMode &&
+        !isSmallScreen()
+      ) {
+        this.handleSelectMode(false);
+      }
+    },
+
+    setLoadingPosition() {
+      let listNode = this.get('listNode');
+
+      if (listNode != null) {
+        if (typeof window === 'undefined') {
+          return;
+        }
+
+        let overlay = listNode.getBoundingClientRect();
+        let viewportHeight = Math.max(
+          document.documentElement.clientHeight,
+          window.innerHeight || 0
+        );
+
+        let overflow = viewportHeight - overlay.height;
+
+        let spinnerHeight =
+          this.get('items.length') === 1
+            ? SMALL_SPINNER_HEIGHT
+            : LARGE_SPINNER_HEIGHT;
+
+        let spinnerPosition =
+          overflow > 0
+            ? (overlay.height - spinnerHeight) / 2
+            : (viewportHeight - overlay.top - spinnerHeight) / 2;
+
+        this.set('loadingPosition', spinnerPosition);
+      }
+    },
+
+    handleSelectAllItemsInStore() {
+      let {
+        onSelectionChange,
+        selectedItems,
+        items,
+        idForItem,
+      } = this.getProperties('onSelectionChange', 'selectedItems', 'items');
```

Comment:
Missing `idForItem` in your `getProperties` here.

Reply:
Great catch! 🙌